specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
Sounds about right to me. For the sake of uniformity I would vote for allowing zero-length date fields. Those equal 0 which in turn means the beginning of the century. Having less exceptions is better, IMHO. Just say that all data types may have a data size of 0 octets, and 0 octets means that all numeric types (signed/unsigned integers, floats) equal 0, the date type equals 0 which is semantically the beginning of the century, the string and binary types equal empty strings/binary elements and for masters it means that there are no child elements.

This is how most programming languages work, too, if they initialize objects.

Less exceptions = less code to test, less things to misunderstand.